core/GlobalSubstitution.h

```diff
@@ -52,18 +54,14 @@ class GlobalSubstitution {
         return substitute(from, allowSameFromTo);
     }
 
-    bool useFastPath() const;
-
 private:
     friend NameRefDebugCheck;
 
     std::vector<NameRef> utf8NameSubstitution;
     std::vector<NameRef> constantNameSubstitution;
     std::vector<NameRef> uniqueNameSubstitution;
-    // set if no substitution is actually necessary
-    bool fastPath;
 
-    const int toGlobalStateId;
+    int toGlobalStateId;
```

Comment:
i missed the motivation for this. was this intentional?

Reply:
Yes, keeping it `const` meant that the default move constructors didn't work. It's an unfortunate consequence, but all the methods on `GlobalSubstitution` are `const` so it didn't feel like the worst solution.